lib/rules/strict.js

```diff
@@ -82,46 +82,46 @@ module.exports = function(context) {
     }
 
     //--------------------------------------------------------------------------
-    // "never" mode
+    // If this is modules, all "use strict" directives are unnecessary.
     //--------------------------------------------------------------------------
 
-    if (mode === "never") {
+    if (context.parserOptions.sourceType === "module") {
```

Comment:
Five rules use `sourceType`:
- no-eval
- no-invalid-this
- no-redeclare
- no-use-before-define
- strict

All use `context.parserOptions.sourceType`; none uses `sourceType` on the Program node. This is probably because 'migrating-to-2.0.0.md' says:

> If you're using `context.ecmaFeatures.modules`, rewrite to check for context.parserOptions.sourceType === "module"`.

PS: 'no-eval' and 'no-invalid-this' are only using `sourceType` to double-check if a scope is in strict mode, which I think is unnecessary.